File: problems/1422_maximum_score_after_splitting_a_string/solution.py
class Solution:
    def maxScore(self, s: str) -> int:
        ones = s.count('1')
        zeros = 0
        best = 0
        for i in range(len(s) - 1):
            if s[i] == '0':
                zeros += 1
            else:
                ones -= 1
            best = max(best, zeros + ones)
        return best

File: problems/1422_maximum_score_after_splitting_a_string/solution.cpp
#include <vector>
#include <string>
#include <unordered_map>
#include <unordered_set>
#include <algorithm>
#include <climits>
#include <queue>
#include <stack>
#include <cmath>
#include <numeric>
#include <set>
#include <map>
using namespace std;

class Solution {
public:
    int maxScore(string s) {
        int ones = count(s.begin(), s.end(), '1');
        int zeros = 0, best = 0;
        for (int i = 0; i < (int)s.size() - 1; i++) {
            if (s[i] == '0') zeros++;
            else ones--;
            best = max(best, zeros + ones);
        }
        return best;
    }
};
